### PR TITLE
feat: Create `LinkButton` component

### PR DESCRIPTION
**Pull request by lucasfp13**
## What's the purpose of this pull request?

This PR intends to create `LinkButton` component on `@faststore/components`.

- [x] Create `LinkButton` component in `@faststore/components`;
- [x] Create `LinkButton` docs;
- [x] Add `LinkButton` to the `Button` docs' _Related components_ section.

## How to test it?

Just run `yarn workspace docs start` and check the `LinkButton` component.

### CODE REVIEW

**Review by emersonlaurentino — APPROVED** [review]

File: packages/components/src/molecules/LinkButton/LinkButton.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      data-testid={testId}
+      data-fs-button-variant={variant}
+      data-fs-button-inverse={inverse}
+      data-fs-button-disabled={disabled}
+      {...otherProps}
+    >
+      {iconPosition === 'left' && <Icon component={icon} />}
+      {children}
+      {iconPosition === 'right' && <Icon component={icon} />}
+    </a>
```
Comment (ArthurTriis1) [inline packages/components/src/molecules/LinkButton/LinkButton.tsx:44]: Shouldn't we use the Link component in this component? In the staters we can't use the frameworks Link in this way.

**Review by renatamottam — APPROVED** [review]
Fast and furious! LGTM :)

File: packages/components/src/molecules/LinkButton/LinkButton.tsx (hunk 1 above)
Comment (lucasfp13) [reply packages/components/src/molecules/LinkButton/LinkButton.tsx]: As discussed before, we'll stick with the anchor tag for now while we don't resolve the style precedence issue. In the future, we will come back to this component.

**Review by ArthurTriis1 — APPROVED** [review]

File: packages/components/src/atoms/List/List.tsx [hunk 2]
```diff
@@ -17,10 +17,8 @@ interface BaseProps {
   marker?: boolean
 }
 
-export type ListProps<T extends ElementType> = PolymorphicComponentPropsWithRef<
-  T,
-  BaseProps
->
+export type ListProps<T extends ElementType = 'ul'> =
```
Comment (eduardoformiga) [inline packages/components/src/atoms/List/List.tsx:20]: is it related to `LinkButton`?

**Review by eduardoformiga — APPROVED** [review]
LGTM!

File: packages/components/src/atoms/List/List.tsx (hunk 2 above)
Comment (lucasfp13) [reply packages/components/src/atoms/List/List.tsx:20]: It was to quick fix the `nextjs.store` build

**Pull request merged**